x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -152,6 +167,26 @@ public void setGrokPattern(String grokPattern) {
         this.grokPattern = grokPattern;
     }
 
+    public long[] getPreferredToCategories() {
+        return preferredToCategories;
+    }
+
+    public void setPreferredToCategories(long[] preferredToCategories) {
+        this.preferredToCategories = preferredToCategories;
+    }
+
+    private void setPreferredToCategories(List<Long> preferredToCategories) {
+        setPreferredToCategories(preferredToCategories.stream().mapToLong(Long::longValue).toArray());
+    }
+
+    public long getNumMatches() {
+        return numMatches;
+    }
+
+    public void setNumMatches(long numMatches) {
+        this.numMatches = numMatches;
```

Comment:
Since the value is being serialized using `writeVLong()` this should throw if passed a negative number.